core/src/test/scala/unit/kafka/log/LogManagerTest.scala

```diff
@@ -1303,6 +1303,29 @@ class LogManagerTest {
         createLeaderAndIsrRequestForStrayDetection(present),
         onDisk.map(mockLog(_))).toSet)
   }
+
+  /**
+   * Test LogManager takes file lock by default and the lock is released after shutdown.
+   */
+  @Test
+  def testLock(): Unit = {
+    val tmpLogDir = TestUtils.tempDir()
+    val tmpLogManager = createLogManager(Seq(tmpLogDir))
+
+    // ${tmpLogDir}.lock is acquired by tmpLogManager
+    val fileLock = new FileLock(new File(tmpLogDir, LogManager.LockFileName))
+    try {
+      assertFalse(fileLock.tryLock())
+    } finally {
+      // ${tmpLogDir}.lock is removed after shutdown
+      tmpLogManager.shutdown()
+      val f = new File(tmpLogDir, LogManager.LockFileName)
+      assertFalse(f.exists())
+
+      fileLock.destroy()
```

Comment:
why we need to call `destroy` here?

Reply:
Originally, I would like to double confirm the file is removed, but I don't think we need it now, because the file should be removed by `LogManager#shutdown`.